app/services/integration/keycloak_service.rb

```diff
@@ -12,26 +12,50 @@ def initialize(integration)
   def call(entry)
     case entry.record
     when Proxy then handle_test
-    when Application then handle_application(entry)
+    when Application then ClientFromApplication.call(entry)
+    when Client then handle_client(entry)
     else handle_rest(entry)
     end
   end
 
-  def handle_application(entry)
+  def handle_client(entry)
     client = build_client(entry)
 
-    unless client.id
-      # Not OAuth
-      return
-    end
-
     if persist?(client)
       persist(client)
     else
       remove(client)
     end
   end
 
+  # Convert Application to Client and trigger new update from the API.
+  # Creates new Client if needed and triggers UpdateJob for it.
+  class ClientFromApplication
+    def self.call(entry)
+      new(entry).call
```

Comment:
_Off topic –_ I've found myself doing the same more than once recently. It would be nice to have a more DSL-ish way to do it, like:
```ruby
define_singleton_sender_for :call
```

```ruby
class Object
  def define_singleton_sender_for(*instance_method_symbols)
    instance_method_symbols.each do |method_symbol|
      define_singleton_method(method_symbol) { |*args| new(*args).send(method_symbol) }
    end
  end
end
```